src/components/table-hoc/reducers/TableRowReducers.ts

```diff
@@ -0,0 +1,69 @@
+import * as _ from 'underscore';
+import {IReduxAction} from '../../../utils/ReduxUtils';
+import {PaginationActions} from '../../navigation/pagination/NavigationPaginationActions';
+import {PerPageActions} from '../../navigation/perPage/NavigationPerPageActions';
+import {ITableRowAddPayload, ITableRowRemovePayload, ITableRowSelectPayload, TableRowActionsType} from '../actions/TableRowActions';
+import {TableHOCComponentType, TableHOCUtils} from '../TableHOCUtils';
+

```

Comment:
this is an interesting way of composing reducers, but i'm wondering if we shouldn't stick to the switch case style since we have this style everywhere?

Reply:
The problem I have with the switch case is that you can't reuse variable name (with const) in multiple cases. Also, the interface of the payload is the same for every reducer which can be confusing